lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py

```diff
@@ -146,14 +177,20 @@
             resource_group: Testing
             virtual_network_name: vnet001
             subnet_name: subnet001
+            ip_configurations:
+                name: ipconfig1
+                public_ip_address_name: publicip001
+                primary: True
 
     - name: Create a network interface with private IP address only (no Public IP)
       azure_rm_networkinterface:
             name: nic001
             resource_group: Testing
             virtual_network_name: vnet001
             subnet_name: subnet001
-            public_ip: no
+            ip_configurations:
```

Comment:
This relates to my comment in the section around public_ip above, if `public_ip` is deprecated, is there a reason why it isn't an option in `ip_configurations`